Add head sampling doc for python (#2485)
open-telemetry/opentelemetry.io: content/en/docs/instrumentation/python/sampling.md

---
title: Sampling
weight: 5
---

[Sampling](/docs/concepts/sampling/) is a process that restricts the amount of

> traces that are generated by a system. the Python SDK offers several

Reviewer: ```suggestion
traces that are generated by a system. The Python SDK offers several
```

[head samplers](/docs/concepts/sampling#head-sampling).

## Default behavior

By default, all spans are sampled, and thus, 100% of traces are sampled. If you
do not need to manage data volume, don't bother setting a sampler.

## TraceIDRatioBased Sampler

> When sampling, the most common head sampler to use is the TraceIdRatioBased
> sampler. It deterministically samples a percentage of traces that you pass in as
> a parameter.

Reviewer: ```suggestion
The most common head sampler to use is [TraceIdRatioBased][].
It deterministically samples a percentage of traces.

[TraceIdRatioBased]: https://opentelemetry-python.readthedocs.io/en/latest/sdk/trace.sampling.html#opentelemetry.sdk.trace.sampling.TraceIdRatioBased
```

Move the link definition to the end of the file.

### Environment Variables

> You can configure the TraceIdRatioBasedSampler with environment variables:

Reviewer: ```suggestion
You can configure a `TraceIdRatioBased` sampler with environment variables:
```

```shell
export OTEL_TRACES_SAMPLER="traceidratio"
export OTEL_TRACES_SAMPLER_ARG="0.1"
```

> This tells the SDK to sample spans such that only 10% of traces get created.

Reviewer: Does it really throttle trace _creation_?

### Configuration in Code

> You can also configure the TraceIdRatioBased sampler in code. Here's an example:

Reviewer: ```suggestion
You can also configure a `TraceIdRatioBased` sampler in code. Here's an example:
```
